xwiki-platform-core/xwiki-platform-oldcore/src/main/java/org/xwiki/store/hibernate/DatabaseProductNameResolved.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.store.hibernate;
+
+import java.util.Optional;
+
+import org.xwiki.component.annotation.Role;
+
+/**
+ * Provide an common identifier from the actual produce name given by the database.
```

Comment:
The doc is a bit poor IMO: I had to look actual implem to understand why you needed that, especially since it's only used for MariaDB right now. I'm a bit wondering if it's not over-engineering right now?

Reply:
> I'm a bit wondering if it's not over-engineering right now?

I don't think so. I don't really have any doubt that we'll eventually have (much) more of those. The only reason why it's only needed for MariaDB right now is that DefaultHibernateAdapterFactory is currently reusing the hacky code of DatabaseProduct#toProduct  but IMO DatabaseProduct will eventually be deprecated, and we should use verious DatabaseProductNameResolver to normalize database identifiers of other DBs too. It makes a lot easier to then register a HibernateAdapter for a specific version of a database.